### PR TITLE
Table of Contents for package admin page.

### PR DESCRIPTION
**Pull request by isoos**
- Deployed to staging for verification.
- The floating part is extremely limited, but provides 80% of the value for very little investment, and we can iterate on it later. When the display is narrow, the table of contents is inlined as part of the admin page.
- Floating example:
image

### CODE REVIEW

**Comment by isoos** [comment]
Note: removed the floating part for now, let's iterate on it separately.

**Comment by jonasfj** [comment]
We should probably deploy this to staging.

**Comment by jonasfj** [comment]
@sigurdm and @szakarias do you have opinions?

File: app/lib/frontend/templates/views/shared/toc.dart [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+/// Renders Table of Contents from [nodes].
+///
+/// The hiearchy of the sections is encoded in [TocNode.children].
+d.Node renderToc(Iterable<TocNode> nodes) {
+  return d.div(
+    classes: ['pub-toc-container'],
+    child: d.div(
+      classes: ['pub-toc'],
+      children: [
+        d.div(child: d.b(text: 'Contents')),
```
Comment (sigurdm) [inline app/lib/frontend/templates/views/shared/toc.dart:16]: Open question. Should we (specifically for the admin page) call this 
```suggestion
        d.div(child: d.b(text: 'Sections')),
```

**Comment by sigurdm** [comment]
There's ...something.. about the layout. I cannot quite put words to it.

But I think it is good enough for starters. We can always iterate on it.

**Review by sigurdm — APPROVED** [review]
This is fine.

I'm OK with keeping 'Contents' just wanted to consider it.

File: app/lib/frontend/templates/views/shared/toc.dart (hunk 1 above)
Comment (isoos) [reply app/lib/frontend/templates/views/shared/toc.dart:16]: I think it is fine to call it `Sections` on any page really. It is not that we can guarantee that a random README actually has a content, only the section headers.

**Pull request merged**